gsitemap.php

```diff
@@ -282,11 +277,9 @@ public function _addLinkToSitemap(&$link_sitemap, $new_link, $lang, &$index, &$i
                 die();
             } else {
                 if ($this->cron) {
-                    header('location: http'.(Configuration::get('PS_SSL_ENABLED') ? 's' : '').'://'.Tools::getShopDomain(false, true).__PS_BASE_URI__.'modules/gsitemap/gsitemap-cron.php?continue=1&token='.Tools::substr(Tools::encrypt('gsitemap/cron'), 0, 10).'&type='.$new_link['type'].'&lang='.$lang.'&index='.$index.'&id='.(int)$id_obj.'&id_shop='.$this->context->shop->id);
+                    Tools::redirect($this->context->link->getBaseLink().'modules/gsitemap/gsitemap-cron.php?continue=1&token='.Tools::substr(Tools::encrypt('gsitemap/cron'), 0, 10).'&type='.$new_link['type'].'&lang='.$lang.'&index='.$index.'&id='.(int)$id_obj.'&id_shop='.$this->context->shop->id);
                 } else {
-                    $admin_folder = str_replace(_PS_ROOT_DIR_, '', _PS_ADMIN_DIR_);
-                    $admin_folder = Tools::substr($admin_folder, 1);
-                    header('location: http'.(Configuration::get('PS_SSL_ENABLED') ? 's' : '').'://'.Tools::getShopDomain(false, true).__PS_BASE_URI__.$admin_folder.'/index.php?tab=AdminModules&configure=gsitemap&token='.Tools::getAdminTokenLite('AdminModules').'&tab_module='.$this->tab.'&module_name=gsitemap&continue=1&type='.$new_link['type'].'&lang='.$lang.'&index='.$index.'&id='.(int)$id_obj.'&id_shop='.$this->context->shop->id);
+                    Tools::redirectAdmin($this->context->link->getBaseLink().'index.php?tab=AdminModules&configure=gsitemap&token='.Tools::getAdminTokenLite('AdminModules').'&tab_module='.$this->tab.'&module_name=gsitemap&continue=1&type='.$new_link['type'].'&lang='.$lang.'&index='.$index.'&id='.(int)$id_obj.'&id_shop='.$this->context->shop->id);
```

Comment:
Watch out, `getBaseLink()` will get you a link to your front office instead of the admin. This can be further simplified if you use `getAdminLink()`:

```php
Tools::redirectAdmin(
  $this->context->link->getAdminLink(
    'AdminModules',
    true,
    array(),
    array(
      'tab_module' => $this->tab
      'module_name' => $this->name,
      'continue' => 1,
      'type' => $new_link['type'],
      'lang' => $lang,
      'index' => $index,
      'id' => (int)$id_obj,
      'id_shop' => $this->context->shop->id,
    )
  )
```

Reply:
Sorry, I don't understand. Can you make a commit, then I will check how exactly you solve this?

Reply:
You can replace the whole line with what @eternoendless sent to you =)

Reply:
When I did that I get an error. I can't remember the error right now. I can reproduce the error and send a screenshot.

Reply:
I added a comma after $this->tab. Please see screenshot:

![screenshot](https://user-images.githubusercontent.com/26626066/43979741-0220c928-9cec-11e8-8ef8-5c8d09d4a8c5.png)

I tried without <code>;</code> too. What am I doing wrong?

Reply:
```
Tools::redirectAdmin(
  $this->context->link->getAdminLink(
    'AdminModules',
    true,
    array(),
    array(
      'tab_module' => $this->tab
      'module_name' => $this->name,
      'continue' => 1,
      'type' => $new_link['type'],
      'lang' => $lang,
      'index' => $index,
      'id' => (int)$id_obj,
      'id_shop' => $this->context->shop->id,
    )
  )
);
```

Missing `);` :D

Reply:
Sorry 🤣

Reply:
Fixed: https://github.com/PrestaShop/gsitemap/pull/53/commits/bf5a4d4d9f4470a455e20920addafb31a871e1d4